### ISSUE TITLE
[JetSmartFilters] "Hide empty"-functionality working incorrect for fields that can have only 1 value

### ISSUE THREAD

**Issue by wesleylancel** [CrocoBlock/suggestions#4596]
**Describe the bug**
When setting "If item empty" to "hide" for "Checkboxes list" filters that are based on a field that can only have 1 value, all other values are hidden as soon as the filter is enabled.

**To Reproduce**
Tested using WooCommerce Products.
- Create two attributes: brand & color. Both can only have 1 value.
- Create "Checkboxes list" filters for both and set "If item empty" to "hide".
- Brand has values A/B/C. Color has values Red/Green/Blue.
- I filter for Brand A. Brand A only has items with color Red or Green. The color filter correctly only shows Red & Green, it hides Blue. However, the brand filter now hides Brand B & C. Obviously these brands aren't present in the resulted items, because an item can only have 1 brand and I'm filtering by Brand A.

**Expected behavior**
Brand filter still shows Brand A, B and C as options Color filter only shows Red & Green.

Basically, to determine what options should be shown in a filter, a query should be run with all filters applied *except* for the one you one want to determine relevant options for. Get unique values for the field of that specific filter from the results and use those as options. Repeat process for all other filters.

**Comment by aalexandr21** [comment]
You need to choose this option

<img width="562" height="578" alt="Image" src="https://github.com/user-attachments/assets/f06a2d29-621b-45d5-8060-2009224e4c72" />
